lib/template-parts.php

```diff
@@ -191,15 +182,55 @@ function filter_rest_wp_template_part_collection_params( $query_params ) {
  */
 function filter_rest_wp_template_part_query( $args, $request ) {
 	if ( $request['theme'] ) {
-		$meta_query   = isset( $args['meta_query'] ) ? $args['meta_query'] : array();
-		$meta_query[] = array(
-			'key'   => 'theme',
-			'value' => $request['theme'],
+		$tax_query   = isset( $args['tax_query'] ) ? $args['tax_query'] : array();
+		$tax_query[] = array(
+			'taxonomy' => 'wp_theme',
+			'field'    => 'slug',
+			'terms'    => $request['theme'],
 		);
 
-		$args['meta_query'] = $meta_query;
+		$args['tax_query'] = $tax_query;
 	}
 
 	return $args;
 }
 add_filter( 'rest_wp_template_part_query', 'filter_rest_wp_template_part_query', 99, 2 );
+
+/**
+ * Filters the post data for a response.
+ *
+ * @param WP_REST_Response $response The response object.
+ * @return WP_REST_Response
+ */
+function filter_rest_prepare_add_wp_theme_slug_and_file_based( $response ) {
+	if ( isset( $response->data ) && is_array( $response->data ) && isset( $response->data['id'] ) ) {
+		$response->data['wp_theme_slug'] = false;
+
+		// Get the wp_theme terms.
+		$wp_themes = wp_get_post_terms( $response->data['id'], 'wp_theme' );
+
+		// If a theme is assigned, add it to the REST response.
+		if ( $wp_themes && is_array( $wp_themes ) ) {
+			$wp_theme_slugs = array_column( $wp_themes, 'slug' );
+
+			$file_based                   = in_array( '_wp_file_based', $wp_theme_slugs, true );
+			$response->data['file_based'] = $file_based;
+
+			$theme_slug = array_values(
+				array_filter(
+					$wp_theme_slugs,
+					function( $slug ) {
+						return '_wp_file_based' !== $slug;
+					}
+				)
+			);
+			if ( $theme_slug ) {
+				$response->data['wp_theme_slug'] = $theme_slug[0];
+			}
+		}
+	}
+
+	return $response;
+}
+add_filter( 'rest_prepare_wp_template', 'filter_rest_prepare_add_wp_theme_slug_and_file_based' );
```

Comment:
@Copons Same problem here. We are using the same function for both... Any idea where we should put this?

Reply:
It's a tricky one.
How do you feel about moving these:

- Taxonomy registration.
- `filter_rest_prepare_add_wp_theme_slug_and_file_based`
- `gutenberg_set_template_post_theme` and `gutenberg_set_template_part_post_theme`
- Anything else related to the **management** (not usage) of the `wp_theme` term that I might have forgotten.

...into a `templates-terms.php` lib? 🤔

Reply:
that makes sense to me. I'm not sure if any other util files exist that would make sense for it

Reply:
Makes sense I suppose... Though tbh I don't know if splitting to more files is better than merging multiple files in 1. 
One could argue that instead of adding yet another file, we should merge the `templates.php` and `template-parts.php` files. There's pros and cons to both approaches, and IMO it's a trivial decision that should not be a blocker :+1: